File: lecture6/coupling.js
// metrics/coupling.js
function calculateCoupling(code) {
    const functionCallRegex = /\b\w+\s*\(.*\)/g;
    const externalFunctionCalls = new Set();
    
    const matches = code.match(functionCallRegex);
    if (matches) {
        matches.forEach(match => {
            externalFunctionCalls.add(match);
        });
    }

    return externalFunctionCalls.size;
}

module.exports = calculateCoupling;

File: lecture6/coupling.py
# metrics/coupling.py

import re

def calculate_coupling(code):
    function_call_pattern = r'\b\w+\s*\(.*?\)'

    matches = re.findall(function_call_pattern, code)
    external_function_calls = set(matches)

    return len(external_function_calls)
